docs/content/en/docs/operator-manual/piped/adding-a-cloud-provider.md

```diff
@@ -77,4 +77,19 @@ See [ConfigurationReference](/docs/operator-manual/piped/configuration-reference
 
 ### Configuring Lambda cloud provider
 
-> TBA
+Adding a Lambda provider requires the region name where Lambda service is running. For authorization, the piped agent gets the credential information from environment variables by default or from a credential file provided via the below `credentialsFile` value.
```

Comment:
Looks like it eventually uses EC2 Instance Role if both environment variables and credential files aren't given. So it's kinder to mention that. (especially as using EC2 Instance Role is highly recommended at our company) And for those who use IAM, it would be nice to show which IAM policy permission is required.

Besides, now the credential chain is a little bit obscure. Hence it would be also cool to be more clear because Piped provides multiple methods of loading credentials.

https://github.com/pipe-cd/pipe/blob/3207f5b54144d1a4d6cc43aef5ded6f4138a3e51/pkg/app/piped/cloudprovider/lambda/client.go#L64-L75

Reply:
So, I'd show you example documents for that. You can ignore this and/or partly refer if needed.

---

You will generally need your AWS credentials to authenticate with Lambda. Piped provides multiple methods of loading these credentials.	
It attempts to retrieve credentials in the following order:	
1. From the environment variables. Available environment variables are `AWS_ACCESS_KEY_ID` or `AWS_ACCESS_KEY` and `AWS_SECRET_ACCESS_KEY` or `AWS_SECRET_KEY`	
1. From the given credentials file.	
1. From the EC2 Instance Role	

Therefore, you don't have to set `credentialsFile` if you use the environment variables or the EC2 Instance Role. Keep in mind the IAM role/user that you use with your Piped must possess the IAM policy permission for `lambda:GetAccountSettings` and `lambda:ListFunctions` and...

Reply:
This is honestly a modified version of ECR document already deleted 🤣

Reply:
Nice 😂 I will use it 🙏